core/Symbols.cc

```diff
@@ -700,29 +701,40 @@ vector<Symbol::FuzzySearchResult> Symbol::findMemberFuzzyMatchConstant(const Glo
                 i++;
             }
             for (const auto scope : candidateScopes) {
-                for (auto member : scope.data(gs)->membersStableOrderSlow(gs)) {
+                scopeBest.clear();
+                for (auto member : scope.data(gs)->members()) {
                     if (member.first.kind() == NameKind::CONSTANT &&
                         member.first.dataCnst(gs)->original.kind() == NameKind::UTF8 && member.second.exists()) {
                         auto thisDistance = Levenstein::distance(
                             currentName, member.first.dataCnst(gs)->original.dataUtf8(gs)->utf8, best.distance);
                         if (thisDistance <= best.distance) {
+                            if (thisDistance < best.distance) {
+                                scopeBest.clear();
+                            }
                             best.distance = thisDistance;
                             best.symbol = member.second;
                             best.name = member.first;
-                            result.emplace_back(best);
+                            scopeBest.emplace_back(best);
                         }
                     }
                 }
+                if (!scopeBest.empty()) {
+                    // NOTE: Iteration order over members is nondeterministic, so we use SymbolId to deterministically
+                    // order the recommendations from this scope.
+                    // We order in decreasing symbol ID order because `result` is later reversed and we want earlier
+                    // ID'd symbols to be recommended first.
+                    fast_sort(scopeBest, [&](auto lhs, auto rhs) -> bool { return lhs.symbol._id > rhs.symbol._id; });
```

Comment:
I don't have the rules for `auto` deduction in lambda lists memorized, but I'm fairly sure that here it resolves to `Symbol::FuzzySearchResult` and we're therefore copy-constructing two items for every comparison.  Even though that copy construction is cheap, I don't think we want to do that.  Can we please use `const auto &` for `lhs` and `rhs` here?

(I realize this is just following the existing practice for the previously-existing sort in this function, but I think it was wrong there, too.)